plugins/module_utils/parse_secrets_v2.py

```diff
@@ -194,6 +194,11 @@ def parse(self):
         secrets = self._get_secrets()
 
         total_secrets = 0  # Counter for all the secrets uploaded
+
+        if secrets == "None" or len(secrets) == 0:
```

Comment:
I think the proper test here is `secrets is None` for a NoneType. Same below. Do you think we should also have tests for the new behaviors?

Reply:
Because we load the yaml with ansible and filter it with from_yaml, it will be a "None" as string, not a NoneType.
I move the null check to the _get_secrets function, and that will always return a list: either 0 length, or the secrets in it.